src/main/scala/strawman/collection/Iterable.scala

```diff
@@ -166,12 +184,19 @@ trait IterableMonoTransforms[+A, +Repr] extends Any {
   def drop(n: Int): Repr = fromIterableWithSameElemType(View.Drop(coll, n))
 
   /** The rest of the collection without its first element. */
-  def tail: Repr = drop(1)
+  def tail: Repr = if (coll.isEmpty) ??? else drop(1)
```

Comment:
I'm not sure we should go so far as to eliminate non-total methods, given their usefulness. Apart from `::` I don't think we have any type that is statically guaranteed to be non-empty.